src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/config/HmppsIncentivesApiExceptionHandler.kt

```diff
@@ -109,6 +110,20 @@ class HmppsIncentivesApiExceptionHandler {
       )
   }
 
+  @ExceptionHandler(IncentiveReviewNotFoundException::class)
+  fun handleRoleNotFoundException(e: IncentiveReviewNotFoundException): ResponseEntity<ErrorResponse?>? {
```

Comment:
probably the method name should be adjusted to something like `handleIncentiveReviewNotFoundException()` :)

I wonder if a method with a more generic `Exception` type could handle all these exceptions that we want to result in a `404 Not Found`, e.g. maybe it's possible to "stack up" several different `@ExceptionHandler` annotations?

e.g.

```Kotlin
@ExceptionHandler(NoPrisonersAtLocationException::class)
@ExceptionHandler(IncentiveReviewNotFoundException::class) 
// etc...
fun handleNotFoundExceptions(e: Exception): ResponseEntity<ErrorResponse?>? {
    log.debug("Not found exception caught: {}", e.message)
    return ResponseEntity
      .status(NOT_FOUND)
      .body(
        ErrorResponse(
          status = NOT_FOUND,
          userMessage = "Not Found: ${e.message}",
          developerMessage = e.message
        )
      )
}
```

although the log message would be less specific. I'm not even sure if that is possible, just a thought, this is working and fine :)

Reply:
updated